rainbow/src/lin_rgba.rs

```diff
@@ -29,3 +31,91 @@ impl LinRgba {
 }
 
 impl_from_into!(LinRgba);
+
+impl Add for LinRgba {
+    type Output = Self;
+
+    fn add(self, rhs: Self) -> Self {
+        Self([
+            self.0[0] + rhs.0[0],
+            self.0[1] + rhs.0[1],
+            self.0[2] + rhs.0[2],
+            self.0[3] + rhs.0[3],
+        ])
+    }
+}
+
+impl<T: en::Float> Add<T> for LinRgba {
+    type Output = Self;
+
+    fn add(self, rhs: T) -> Self {
+        util::map_all(self.0, |lhs| (lhs.cast::<T>() + rhs).to_f32()).into()
+    }
+}
+
+impl Sub for LinRgba {
+    type Output = Self;
+
+    fn sub(self, rhs: Self) -> Self {
+        Self([
+            self.0[0] - rhs.0[0],
+            self.0[1] - rhs.0[1],
+            self.0[2] - rhs.0[2],
+            self.0[3] - rhs.0[3],
+        ])
+    }
+}
+
+impl<T: en::Float> Sub<T> for LinRgba {
+    type Output = Self;
+
+    fn sub(self, rhs: T) -> Self {
+        util::map_all(self.0, |lhs| (lhs.cast::<T>() - rhs).to_f32()).into()
+    }
+}
+
+impl Mul for LinRgba {
+    type Output = Self;
+
+    fn mul(self, rhs: Self) -> Self {
+        Self([
+            self.0[0] * rhs.0[0],
+            self.0[1] * rhs.0[1],
+            self.0[2] * rhs.0[2],
+            self.0[3] * rhs.0[3],
+        ])
+    }
+}
+
+impl<T: en::Float> Mul<T> for LinRgba {
+    type Output = Self;
+
+    fn mul(self, rhs: T) -> Self {
+        util::map_all(self.0, |lhs| (lhs.cast::<T>() * rhs).to_f32()).into()
+    }
+}
+
+impl Div for LinRgba {
+    type Output = Self;
+
+    fn div(self, rhs: Self) -> Self {
+        Self([
+            self.0[0] / rhs.0[0],
+            self.0[1] / rhs.0[1],
+            self.0[2] / rhs.0[2],
+            self.0[3] / rhs.0[3],
+        ])
+    }
+}
+
+impl<T: en::Float> Div<T> for LinRgba {
+    type Output = Self;
+
+    fn div(self, rhs: T) -> Self {
+        util::map_all(self.0, |lhs| (lhs.cast::<T>() / rhs).to_f32()).into()
+    }
+}
+
+pub fn Lerp<F: en::Float>(start: LinRgba, end: LinRgba, factor: F) -> LinRgba {
```

Comment:
Do you think making this a method would obscure the start/end too much?

Reply:
hmm, good question. I do think a free-standing function is a bit more clear, but I'm not opposed to a method.